myceli/src/shipper.rs

```diff
@@ -75,17 +85,26 @@ impl<T: Transport + Send + 'static> Shipper<T> {
         })
     }
 
+    // Single point of receiving messages off the receive channel
     pub fn receive_msg_loop(&mut self) {
         loop {
             if let Ok((message, sender_addr)) = self.receiver.recv() {
-                if let Err(e) = self.receive(message, &sender_addr) {
+                if let Err(e) = self.process_msg(message, &sender_addr) {
                     error!("{e:?}");
                 }
             }
         }
     }
 
-    pub fn receive(&mut self, message: DataProtocol, sender_addr: &str) -> Result<()> {
+    // Examine a received message and take appropriate action
+    pub fn process_msg(&mut self, message: DataProtocol, sender_addr: &str) -> Result<()> {
+        // Find a reasonable target address to respond to by either using our radio_address
+        // or using the sender_addr if no radio address is set
+        let target_addr = if let Some(radio_address) = &self.radio_address {
+            radio_address.to_owned()
```

Comment:
Sorry for still being on basic questions - under what cases is radio_address Some vs. None?

Reply:
I see - it's a passthrough of a configuration point, and None means I didn't write it in my config file.

So if someone asks me to transmit a thing, I'll always send it to the configured point if one is configured, but if not I'll send it to the address that asked for it to be sent?

I think that makes sense. But don't a lot of these requests have a target address as a parameter? Does that take priority?

Reply:
You are right that many of the requests do take a target address as a parameter. The radio address is mainly there to support responses to requests which don't take a target address, but in reality (for this specific demo) it is generally the correct address to send responses to. 

The main caveat here is when using the `controller` and when the `radio_address` is set...but both in actual testing and local testing we've had access to the `myceli` logs, which has allowed us access to any response information that was needed.